tools/mmp/Tuning.cs

```diff
@@ -112,8 +112,10 @@ public static void Process (LinkerOptions options, out MonoMacLinkContext contex
 					if (a != null)
 						message.AppendLine ($"\tAssembly: `{a}`");
 				}
+				if (e.InnerException is AssemblyResolutionException res)
+					message.AppendLine ("Attempting to resolve: " + res.AssemblyReference?.Name);
```

Comment:
It looks like the _newest_ linker code has wrapped the original exception :(
See line 89
https://github.com/xamarin/xamarin-macios/pull/6598/files#diff-63408d1add0e85dbd60bdcfc4dfff73dR89

This should have been a `MM2002` error - and it's likely (some/all?) other cases will also miss their handlers.

Note: the same linker behaviour change likely affects `mtouch` too

Reply:
I suspect fixing that will be a tad more invasive.